goth/assertions/monitor.py

```diff
@@ -112,11 +112,20 @@ def add_assertion(
     ) -> Assertion:
         """Add an assertion function to this monitor."""
 
-        assertion = Assertion(self._events, assertion_func, name=name)
-        assertion.start()
+        assertion = Assertion(assertion_func, name=name)
+        self.attach_assertion(assertion, log_level)
+        return assertion
+
+    def attach_assertion(
```

Comment:
I'm worried it might be somewhat confusing to have both these functions (i.e. `add_assertion` and `attach_assertion`) in the monitor's public API. Maybe this is a good use-case for `typing.overload`?

Reply:
Don't be! I fixed this as you suggested in 9dff362.